package/zmq_adapter/src/filter_sbp.c

```diff
@@ -41,6 +46,16 @@ static int process_rule(filter_sbp_rule_t *rule)
 void filter_sbp_init(void *filter_sbp_state, const char *filename)
 {
   filter_sbp_state_t *s = (filter_sbp_state_t *)filter_sbp_state;
+  s->config_file = filename;
```

Comment:
Copy filename to a local buffer instead of assuming it persists outside of this function.